### PR TITLE
For you to follow 453

### PR DESCRIPTION
**Pull request by msamgan**
with update: show the questions only posted by the people you are following. as requested by @MrPunyapal

### CODE REVIEW

File: routes/web.php [hunk 1]
```diff
@@ -19,7 +19,8 @@
 Route::view('/about', 'about')->name('about');
 
 Route::view('/', 'home/feed')->name('home.feed');
-Route::view('/for-you', 'home/questions-for-you')->name('home.for_you');
+Route::get('/for-you', fn () => redirect()->route('home.following'))->name('home.for_you');
```
Comment (MrPunyapal) [inline routes/web.php:22]: just use
```php
  Route::permanentRedirect('/for-you', '/following');
```

File: tests/Http/ForYouRedirectTest.php [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<?php
+
+declare(strict_types=1);
+
+test('/for-you redirects to /following', function () {
```
Comment (MrPunyapal) [inline tests/Http/ForYouRedirectTest.php:5]: move it to following tests

File: tests/Http/Home/FollowingTest.php [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+        ->assertSee('Log in or sign up to access personalized content');
+});
+
+it('does increment views', function () {
+    Queue::fake(IncrementViews::class);
+
+    $this->actingAs(User::factory()->create());
+    $this->get(route('home.following'));
+
+    Queue::assertPushed(IncrementViews::class);
```
Comment (MrPunyapal) [inline tests/Http/Home/FollowingTest.php:33]: no need remove this test

File: tests/Unit/Feeds/QuestionsForYouTest.php [hunk 4]
```diff
@@ -30,7 +30,7 @@
         'is_reported' => false,
     ]);
 
-    $builder = (new QuestionsForYouFeed($likerUser))->builder();
+    $builder = (new QuestionsFollowingFeed($likerUser))->builder();
```
Comment (MrPunyapal) [inline tests/Unit/Feeds/QuestionsForYouTest.php:33]: update the tests too

File: tests/Unit/Livewire/Home/FollowingTest.php [hunk 5]
```diff
@@ -26,7 +26,7 @@
 
     $question->likes()->save($like);
```
Comment (MrPunyapal) [inline tests/Unit/Livewire/Home/FollowingTest.php:27]: tests are still seems for for-you feed

**Review by MrPunyapal — CHANGES_REQUESTED** [review]

File: tests/Http/ForYouRedirectTest.php (hunk 2 above)
Comment (msamgan) [reply tests/Http/ForYouRedirectTest.php:5]: can you please elaborate on this one ?

File: tests/Unit/Feeds/QuestionsFollowingTest.php [hunk 6]
```diff
@@ -19,50 +18,19 @@
         'is_reported' => false,
     ]);
 
-    Like::factory()->create([
-        'user_id' => $likerUser->id,
-        'question_id' => $questionWithLike->id,
-    ]);
+    $likerUser->following()->attach($userTo->id);
```
Comment (MrPunyapal) [inline tests/Unit/Feeds/QuestionsFollowingTest.php:21]: name is not clear

File: tests/Unit/Livewire/Home/FollowingTest.php [hunk 7]
```diff
@@ -2,35 +2,15 @@
...
-
-    $component = Livewire::actingAs($user)->test(QuestionsForYou::class);
-
-    $component
-        ->assertDontSee('We haven\'t found any questions that may interest you based on the activity you\'ve done on Pinkary.')
-        ->assertSee($questionContent);
+    $response->assertRedirect('/following');
 });
 
 test('do not renders questions without likes of authenticated user', function () {
```
Comment (MrPunyapal) [inline tests/Unit/Livewire/Home/FollowingTest.php:16]: tests still needs update (remove all likes stuff)

**Review by MrPunyapal — CHANGES_REQUESTED** [review]

File: tests/Unit/Livewire/Home/ForYouTest.php [hunk 8]
```diff
@@ -1,73 +0,0 @@
...
-        'from_id' => $user->id,
-        'to_id' => $user->id,
-    ]);
-
-    $component = Livewire::actingAs($user)->test(QuestionsForYou::class);
-
-    $component
-        ->assertSee('found any questions that may interest you based on the activity')
-        ->assertDontSee($questionContent);
-});
```
Comment (MrPunyapal) [inline tests/Unit/Livewire/Home/ForYouTest.php:53]: Can you write similar tests for the following? 🤔

File: tests/Unit/Livewire/Home/FollowingTest.php [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+use App\Livewire\Home\QuestionsFollowing;
+use App\Models\Question;
+use App\Models\User;
+use Livewire\Livewire;
+
+test('/for-you redirects to /following', function () {
+    $response = $this->get('/for-you');
+
+    $response->assertRedirect('/following');
+});
```
Comment (MrPunyapal) [inline tests/Unit/Livewire/Home/FollowingTest.php:14]: This is a feature test not a unit so can you put it into a feature test? 🤔

**Review by MrPunyapal — CHANGES_REQUESTED** [review]
Last few changes 😅

**Review by MrPunyapal — APPROVED** [review]

**Pull request merged**